src/datanode/src/region_server.rs

```diff
@@ -140,6 +152,12 @@ impl RegionServer {
         region_id: RegionId,
         request: RegionRequest,
     ) -> Result<RegionResponse> {
+        if let Some(p) = &self.inner.parallelism {
+            let permit = p.acquire().await?;
+            let res = self.inner.handle_request(region_id, request).await;
+            drop(permit);
+            return res;
+        }
```

Comment:
Looks like this semaphore also applies to `RegionPutRequest` whilst the purpose is to limit concurrent queries. We need to distinguish different request types.

Reply:
should i put the rate limiting on `handle_read` and `handle_remote_read` that does datafusion queries? seems like `handle_requests` method is more of writes and region ddl changes.